src/Equinor.ProCoSys.IPO.Command/InvitationCommands/EditInvitation/EditInvitationCommandHandler.cs

```diff
@@ -100,124 +103,73 @@ public async Task<Result<string>> Handle(EditInvitationCommand request, Cancella
             return new SuccessResult<string>(invitation.RowVersion.ConvertToString());
         }
 
-        private async Task UpdateMcPkgScopeAsync(Invitation invitation, IList<string> mcPkgNos, string projectName)
+        private async Task<List<McPkg>> GetMcPkgScopeAsync(IList<string> mcPkgNos, string projectName)
         {
-            var existingMcPkgScope = invitation.McPkgs;
-            var excludedMcPkgs = existingMcPkgScope.Where(mc => !mcPkgNos.Contains(mc.McPkgNo)).ToList();
-            RemoveMcPkgs(invitation, excludedMcPkgs);
-
-            var existingMcPkgNos = existingMcPkgScope.Select(mc => mc.McPkgNo);
-            var newMcPkgs = mcPkgNos.Where(mcPkgNo => !existingMcPkgNos.Contains(mcPkgNo)).ToList();
-            if (newMcPkgs.Count > 0)
+            if (mcPkgNos.Count > 0)
             {
-                await AddMcPkgsAsync(invitation, newMcPkgs, projectName,
-                    existingMcPkgScope.Count > 0 ? existingMcPkgScope.First().SystemSubString : null);
-            }
-        }
+                var mcPkgsFromMain =
+                    await _mcPkgApiService.GetMcPkgsByMcPkgNosAsync(_plantProvider.Plant, projectName, mcPkgNos);
 
-        private void RemoveMcPkgs(Invitation invitation, IEnumerable<McPkg> excludedMcPkgs)
-        {
-            foreach (var mcPkgToDelete in excludedMcPkgs)
-            {
-                invitation.RemoveMcPkg(mcPkgToDelete);
-                _invitationRepository.RemoveMcPkg(mcPkgToDelete);
```

Comment:
@janid1967 just a comment since I have not touched this file:
Consider if changes should be done to InvitationRepository and InvitationRepositoryTests since RemoveMcPkg and RemoveCommPkg is not used from handler any more

Reply:
Unused methods in InvitationRepository are removed